### ISSUE TITLE
Support deletion in PathMappingRepository

### ISSUE THREAD

**Issue by webmozart** [puli/issues#105]
The `PathMappingRepository` should support deletion of existing mappings. For example, if `/app` is mapped to `res`, it should be possible to delete this mapping:

```php
$repo->add('/app', new DirectoryResource(...));

// fails
$repo->remove('/app/resource.txt');

// succeeds
$repo->remove('/app');
```

cc @tgalopin

**Comment by tgalopin** [comment]
Can you assign this, https://github.com/puli/issues/issues/105 and https://github.com/puli/issues/issues/90 to me so I can retrieve them easily? I'll work on these as soon as possible.

This one seems quite easy, I'll work on it this night.

**Comment by webmozart** [comment]
I can only assign organization members. So if you accept my invitation, I can assign you :)

**Comment by tgalopin** [comment]
Well, okay :D . Thanks!

**Comment by webmozart** [comment]
Fixed in puli/repository#46
